src/layout/core/tree.rs

```diff
@@ -40,15 +41,20 @@ pub enum TreeError {
     /// There was an error in the graph, an invariant of one of the
     /// functions were not held, so this might be an issue in the Tree.
     PetGraph(GraphError),
-    /// An error occured while trying to focus on a container
-    Focus(FocusError)
+    /// An error occurred while trying to focus on a container
+    Focus(FocusError),
+    /// An error occurred while trying to move a container
+    Movement(MovementError),
+    /// The tree was (true) or was not (false) performing an action,
+    /// but the opposite value was expected.
+    PerformingAction(bool)
```

Comment:
This is confusing and stupid, the only case this will matter is if the user is doing something when we are trying to do something else.

Change this to `UserPerformingAction`